rustyms-py/src/lib.rs

```diff
@@ -0,0 +1,1262 @@
+    /// Returns
+    /// -------
+    /// float
+    ///
+    #[getter]
+    fn experimental_mz(&self) -> f64 {
+        self.0.experimental_mz.value
+    }
+
+    /// The intensity of the peak.
+    ///
+    /// Returns
+    /// -------
+    /// float
+    ///
+    #[getter]
+    fn intensity(&self) -> f64 {
+        self.0.intensity
+    }
+
+    /// The charge of the peak.
+    ///
+    /// Returns
+    /// -------
+    /// int
+    ///
+    #[getter]
+    fn charge(&self) -> i16 {
+        self.0.charge.value as i16
+    }
+
+    /// The annotation of the peak, if present.
+    ///
+    /// Returns
+    /// -------
+    /// list[Fragment]
+    ///
+    #[getter]
+    fn annotation(&self) -> Vec<Fragment> {
+        self.0
+            .annotation
+            .iter()
+            .map(|x| Fragment(x.clone()))
+            .collect()
+    }
+}
+
+/// A raw spectrum (meaning not annotated yet)
+///
+/// Parameters
+/// ----------
+/// title : str
+///     The title of the spectrum.
+/// num_scans : int
+///     The number of scans.
+/// rt : float
+///     The retention time.
+/// precursor_charge : float
+///     The found precursor charge.
+/// precursor_mass : float
+///     The found precursor mass.
+/// mz_array : list[float]
+///     The m/z values of the peaks.
+/// intensity_array : list[float]
+///     The intensities of the peaks.
+///
+/// Returns
+/// -------
+/// RawSpectrum
+///
+#[pyclass]
+pub struct RawSpectrum(rustyms::RawSpectrum);
+
+#[pymethods]
+impl RawSpectrum {
+    /// Create a new raw spectrum.
+    #[new]
+    fn new(
+        title: &str,
+        num_scans: u64,
+        rt: f64,
+        precursor_charge: f64,
+        precursor_mass: f64,
+        mz_array: Vec<f64>,
+        intensity_array: Vec<f64>,
+    ) -> Self {
+        RawSpectrum(rustyms::RawSpectrum {
+            title: title.to_string(),
+            num_scans,
+            rt: rustyms::system::Time::new::<rustyms::system::s>(rt),
+            charge: rustyms::system::Charge::new::<rustyms::system::e>(precursor_charge),
+            mass: rustyms::system::Mass::new::<rustyms::system::dalton>(precursor_mass),
+            spectrum: mz_array
+                .into_iter()
+                .zip(intensity_array.into_iter())
+                .map(|(mz, i)| rustyms::spectrum::RawPeak {
+                    charge: rustyms::system::Charge::new::<rustyms::system::e>(1.0),
+                    mz: rustyms::system::MassOverCharge::new::<rustyms::system::mz>(mz),
+                    intensity: i,
+                })
+                .collect(),
+            intensity: None,
+            sequence: None,
+            raw_file: None,
+            raw_scan_number: None,
+            raw_index: None,
+            sample: None,
+            period: None,
+            cycle: None,
+            experiment: None,
+            controller_type: None,
+            controller_number: None,
+        })
+    }
+
+    // fn __str__(&self) -> String {
+    //     todo!()
```

Comment:
Maybe the simplest would be to use Debug. Otherwise this could be left empty until I build MGF export at some point.